CareKitStore/CareKitStore/Protocols/CoreData/OCKCoreDataStoreProtocol.swift

```diff
@@ -178,6 +178,7 @@ extension OCKCoreDataStoreProtocol {
             guard let current = currentVersions.first(where: { $0.id == value.id }) else {
                 throw OCKStoreError.invalidValue(reason: "No matching object could be found for id: \(value.id)")
             }
+            current.logicalClock = Int64(context.clockTime)
```

Comment:
I'm inferring that this was added to ensure that both the old and new version of the updated object are captured in the next revision. Do you recall why does the old version need to be captured?

Reply:
I'm trying to remember, but I believe it had something to do with the old versions "nextVersion" being updated, but not sent to the remote. This ensures that the linkList between versions is updated on the remote side. I think I added a testcase for this also, but will look back through

When this wasn't here, I had to add code on the remote side to repair previous and next version link list because I didn't receive the update from the OCKStore

Reply:
That's what I suspected! I think we would prefer to leave it up to the remote to update the previous <--> next link instead of syncing over both the old and new version!

Reply:
I can revert, but my intuition is telling me remotes may mess this up since locally a change was made, but never pushed to the server. What I can see happening is if a remote doesn't link properly and a user wants to restore their device from the Cloud then it would be up to the OCKStore to do the same repair type of repair the remote should have done, but on the client side during pullRevisions by checking previousVersion, nextVersion and establishing the link.